packages/expo-cli/src/commands/eject/Eject.ts

```diff
@@ -129,18 +140,45 @@ export async function ejectAsync(projectRoot: string, options) {
   }
 }
 
-async function ejectToBareAsync(projectRoot, options) {
+function isStringOrNumber(value: any): value is string | number {
+  return typeof value === 'string' || typeof value === 'number';
+}
+
+function ensureDependenciesMap(value: any): DependenciesMap {
+  if (typeof value !== 'object') {
+    throw new Error(`Dependency map is invalid, expected object but got ${typeof value}`);
+  }
+
+  const outputMap: DependenciesMap = {};
+  if (!value) return outputMap;
+
+  for (const key of Object.keys(value)) {
+    if (key in value) {
+      if (isStringOrNumber(value)) {
```

Comment:
`number` is not a valid version number.

The JSON I gave in the earlier [comment](https://github.com/expo/expo-cli/pull/1124#discussion_r334874750):
```json
{"dependencies": {"lodash": 4.0, "react-native": "0.61.2"} }
```
was just an example of *invalid* package.json content that should throw an error. Sorry if the comment was unclear!